jmxtrans-core/src/main/java/com/googlecode/jmxtrans/guice/JmxTransModule.java

```diff
@@ -83,7 +87,7 @@ protected void configure() {
 				.toInstance(getObjectPool(new SocketFactory(), SocketFactory.class.getSimpleName()));
 		bind(new TypeLiteral<GenericKeyedObjectPool<SocketAddress, DatagramSocket>>(){})
 				.toInstance(getObjectPool(new DatagramSocketFactory(), DatagramSocketFactory.class.getSimpleName()));
-		bind(KeyedObjectPool.class).annotatedWith(Names.named("mbeanPool"))
+		bind(new TypeLiteral<KeyedObjectPool<JmxConnectionProvider, JMXConnection>>(){}).annotatedWith(Names.named("mbeanPool"))
```

Comment:
Well I don't know, but without that Guice complains that he can not find a bean of type ... and named mbeanPool, in short it doesn't work at all. Upgrading Jackson may have introduced some changes in dependencies.